modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -23,6 +37,317 @@
+        """
+        columns_length = len(columns)
+        if columns_length == 0:
+            return [], []
+        num_partitions = NPartitions.get()
+        column_splits = (
+            columns_length // num_partitions
+            if columns_length % num_partitions == 0
+            else columns_length // num_partitions + 1
+        )
+        col_partitions = [
+            columns[i : i + column_splits]
+            for i in range(0, columns_length, column_splits)
+        ]
+        column_widths = [len(c) for c in col_partitions]
+        return col_partitions, column_widths
+
+    @classmethod
+    def build_dtypes(cls, partition_ids, columns):
+        """
+        Compute common for all partitions `dtypes` for each of the DataFrame column.
+
+        Parameters
+        ----------
+        partition_ids : list
+            Array with references to the partitions data.
+        columns : list
+            List of columns that should be read from file.
+
+        Returns
+        -------
+        dtypes : pandas.Series
+            Series with dtypes for columns.
+        """
+        dtypes = pandas.concat(cls.materialize(list(partition_ids)), axis=0)
+        dtypes.index = columns
+        return dtypes
+
+    @classmethod
+    def build_query_compiler(cls, path, columns, **kwargs):
+        """
+        Build query compiler from deployed tasks outputs.
+
+        Parameters
+        ----------
+        path : str, path object or file-like object
+            Path to the file to read.
+        columns : list
+            List of columns that should be read from file.
+        **kwargs : dict
+            Parameters of deploying read_* function.
+
+        Returns
+        -------
+        new_query_compiler : BaseQueryCompiler
+            Query compiler with imported data for further processing.
+        """
+        col_partitions, column_widths = cls.build_columns(columns)
+        partition_ids = cls.call_deploy(path, col_partitions, **kwargs)
+        index, needs_index_sync = cls.build_index(path, kwargs["storage_options"])
+        remote_parts = cls.build_partition(partition_ids, column_widths)
+        if len(partition_ids) > 0 and len(partition_ids[0]) > 0:
+            first_row = partition_ids[0]
+            dtypes = cls.build_dtypes(
+                [dtype_and_partition[1] for dtype_and_partition in first_row], columns
+            )
+            row_lengths = [part.length() for part in remote_parts.T[0]]
+        else:
+            dtypes = None
+            row_lengths = None
+        frame = cls.frame_cls(
+            remote_parts,
+            index,
+            columns,
+            row_lengths=row_lengths,
+            column_widths=column_widths,
+            dtypes=dtypes,
+        )
+        new_query_compiler = cls.query_compiler_cls(frame)
+        if needs_index_sync:
+            frame.synchronize_labels(axis=0)
+        return new_query_compiler
```

Comment:
I would rather swap the statements (for mostly aesthetical purposes of finishing `frame` construction before building a query compiler from it), and do smth like
```suggestion
        if needs_index_sync:
            frame.synchronize_labels(axis=0)
        return cls.query_compiler_cls(frame)
```